### ISSUE TITLE
Running Laravel with PeachPie

### ISSUE THREAD

**Issue by petrsvihlik** [peachpiecompiler/peachpie#312]
Hey guys,
so I decided to give PeachPie a try and run a [sample application for Kentico Cloud](https://github.com/Kentico/kentico-cloud-sample-app-php) that's built on top of the Laravel framework.
I ran into numerous error, so I stripped it down and ran just a basic Laravel boilerplate. Here's what I did in a terminal:

```
composer global require "laravel/installer"
laravel new myproject
``` 
This works fine if I run `php artisan serve`. But when I try to compile it with PeachPie I get two errors:
- one's described in #235 (or rather in https://github.com/mockery/mockery/issues/897)
- the other comes from `phpunit`
> Severity	Code	Description	Project	File	Line	Suppression State
Error	PHP2014	Syntax error: unexpected token '{'	peachpie_web	C:\Users\user\source\repos\peachpie_web\peachpie_web\peachpie_web\todo\vendor\phpunit\php-code-coverage\tests\_files\Crash.php	2	

I realize that the problems might not be on your side (I really didn't have time to dig deeper) but considering the popularity of Laravel and PHPUnit, I think this is something worth looking into.

Do you have any insights or recommendations? Did I overlook something?

**Comment by Remo** [comment]
Did you try to ignore the code? I had to do something similar with a different project, check this https://github.com/peachpiecompiler/peachpie/issues/289

**Comment by jakubmisek** [comment]
Hi, all the test scripts in general should be excluded from the compilation. Your code does not reference `phpunit` package so none of the tests can be compiled.

Edit the project `.msbuildproj` file and within the `<Compile ... />` tag add something like `Exclude="**/tests/**;**/test/**"` or similar.

Then you should be able to proceed to next issues : )

**Comment by jakubmisek** [comment]
Lets close the issue for now. The title is little bit too generic.

A **common recommendation** is to **exclude test files** unless you actually want to compile and run tests (then you should include phpunit phar file and be careful to not publish this on production)

`PHP2014 Syntax error` is caused by a syntax error in source files. Unlike PHP, PeachPie recognizes the error in compile time and does not allow such project to be compiled successfully. In this particular case, the file is a test that should cause a runtime error which is pointless in our case.

**Comment by bfistein** [comment]
@amirvenus this issue is only 11 days old since being closed, so since then there haven't been any updates. We are currently not specifically targeting Laravel (but planning to do it sometime soon). Have you tried it? You can always test it and post the exception here in an issue and we'll fix that relatively quickly.

**Comment by petrsvihlik** [comment]
@jakubmisek @Remo thanks for the suggestion. I tried it last week. The general idea of excluding `**/tests/**` is not bad, however there are some projects that don't follow any convention with regards to their test structure... If I remember correctly, some libraries have their tests stored in a `phpunit` folder... which makes distinguishing between test classes and phpunit test framework impractical (if not literally impossible). that poses a problem for libraries that depend on phpunit but don't have tests stored in a predictable location....so after an hour or so, I gave up :)

I decided to take a step back and try something simpler. I'm playing around with PHP/C# interop. I'm able to compile my code but I'm running into other issues. See #335.
